### PR TITLE
Fix the linux install based on testing.

### PR DESCRIPTION
**Pull request by csilvers**
## Summary:
It looks like the linux install of khan-dotfiles has bitrotted a bit.
This fixes it up based on a new test regime that occured to me: just
use a docker container for testing!  I wrote up how to do this in the
README.

The main changes I made:
* Add a few packages that might not be installed by default
* Make sure /usr/bin/python is installed
* Make sure the git ssh key is installed correctly

Issue: none

## Test plan:
As described in the README.  After running `make` I could run
`make serve` (in webapp) with success!

### CODE REVIEW

File: linux-setup.sh [hunk 1]
```diff
@@ -91,7 +91,7 @@ install_packages() {
     # This is needed to get the add-apt-repository command.
     # apt-transport-https may not be strictly necessary, but can help
     # for future updates.
-    sudo apt-get install -y software-properties-common apt-transport-https
+    sudo apt-get install -y software-properties-common apt-transport-https wget
```
Comment (dbraley) [inline linux-setup.sh:94]: 👍🏻

File: linux-setup.sh [hunk 2]
```diff
@@ -250,7 +253,8 @@ EOF
 
     # Not needed for Khan, but useful things to have.
     sudo apt-get install -y ntp abiword diffstat expect gimp \
-        mplayer netcat netpbm screen w3m vim emacs google-chrome-stable
+         mplayer netcat sysdig iftop tcpflow netpbm screen w3m \
```
Comment (dbraley) [inline linux-setup.sh:256]: Cool!

File: shared-functions.sh [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
+# for replacing previous input prompts with warnings
+warn() {
+    printf "\r\033[2K  [\033[0;33mWARN\033[0m] $1\n"
+}
+
+# for replacing previous prompts with errors
+error() {
+    printf "\r\033[2K  [\033[0;31mFAIL\033[0m] $1\n"
+}
+
```
Comment (dbraley) [inline shared-functions.sh:34]: 👍🏻

File: linux-functions.sh [hunk 4]
```diff
@@ -1,310 +0,0 @@
-#!/bin/sh
-# This is largely a copy of linux-setup.sh
-# TODO(ericbrown): Reduce code redundancy by refactoring to share code
-
-install_java() {
```
Comment (dbraley) [inline linux-functions.sh:5]: I don't see where these got moved to, where they redundant?

**edit** Ok, cool, most if it was already in `linux-setup.sh`, and most of the important bits that wasn't was ported there. Looks like we lost installing the slack app 👍🏻  but I think I honestly prefer using the web version on linux, I'm more than ok with leaving it up to individual ubuntu users to determine if they want to install it themselves.

**Review by dbraley — APPROVED** [review]
I love the testing in a container approach! Unfortunately, it doesn't seem to run all the way through for me. I get `E: Unable to locate package google-chrome-stable` when run in a container (at least on my m1, though that doesn't seem _likely_ to be arch related, but I suppose it could be). Interestingly, I seem to get an error upgrading sysdig (specifically the sysdig-dkms` dependency) on my linux machine, but that one is because of a failing build for the arch, so I don't expect it's related to this, but is making it hard for me to test the 'rerun on a current machine' case.

I don't have any issue with anything done here, it all looks good. If we can resolve the import issues, or if you don't have them, then I'm ok moving forward.

**Comment by dbraley** [comment]
After skipping `google-chrome-stable` inside the ubuntu container, I'm now hitting the `sysdig-dkms` problem there too, on my M1. I still suspect it's an ephemeral thing, I'll see if I can't force it back to an older version.

Because these are optional installs, we may want to consider just emitting a warning and moving on when we hit a problem with these installs.

**Comment by csilvers** [comment]
> After skipping `google-chrome-stable` inside the ubuntu container, I'm now hitting the `sysdig-dkms` problem there too, on my M1. I still suspect it's an ephemeral thing, I'll see if I can't force it back to an older version.
> 
> Because these are optional installs, we may want to consider just emitting a warning and moving on when we hit a problem with these installs.

Weird, did you try running the exact test plan I had in the commit message, or something different?

Let me just take out sysdig, I don't think it's very valuable.

**Comment by dbraley** [comment]
> Weird, did you try running the exact test plan I had in the commit message, or something different?

Same plan, other than I also check out the `linux-fixes`.

**Comment by csilvers** [comment]
> > Weird, did you try running the exact test plan I had in the commit message, or something different?
> 
> Same plan, other than I also check out the `linux-fixes`.

Aha!  I was able to repro.  The problem is gnupg wasn't installed when trying to verify the key.  I fixed that.

**Pull request merged**